src/ring_theory/localization.lean

```diff
@@ -2064,19 +2070,33 @@ end localization
 
 open is_localization
 
-/-- If `R` is a field, then localizing at a submonoid not containing `0` adds no new elements. -/
-lemma localization_map_bijective_of_field
-  {R Rₘ : Type*} [comm_ring R] [is_domain R] [comm_ring Rₘ]
-  {M : submonoid R} (hM : (0 : R) ∉ M) (hR : is_field R)
+open_locale non_zero_divisors
+
+/-- If `R` is a field, then localizing at a submonoid smaller than the non-zero divisors
+    adds no new elements. -/
+lemma localization_map_bijective_of_field'
+  {R Rₘ : Type*} [comm_ring R] [comm_ring Rₘ]
+  {M : submonoid R} (hM : M ≤ R⁰) (hR : is_field R)
```

Comment:
"a submonoid smaller than the non-zero divisors" -- that's just the same as anything that doesn't contain `0`, right?

Reply:
some of the lemmas I added are now spurious, I hope that's fine because I do think they're still genuinely useful